rules/terraformrules/terraform_required_providers.go

```diff
@@ -126,7 +127,14 @@ func (r *TerraformRequiredProvidersRule) Check(runner *tflint.Runner) error {
 			continue
 		}
 
-		val, diags := provider.Expr.Value(nil)
+		val, diags := provider.Expr.Value(&hcl.EvalContext{
+			Variables: map[string]cty.Value{
+				// configuration_aliases can declare additional provider instances
+				// required provider "foo" could have: configuration_aliases = [foo.a, foo.b]
+				// @see https://www.terraform.io/language/modules/develop/providers#provider-aliases-within-modules
+				name: cty.DynamicVal,
+			},
+		})
```

Comment:
Interesting. FWIW, Terraform core uses `hcl.ExprMap`, so it's probably best to do so.
https://github.com/hashicorp/terraform/blob/v1.2.6/internal/configs/provider_requirements.go#L78